org.eclipse.m2e.core/src/org/eclipse/m2e/core/internal/MavenArtifactIdentifier.java

```diff
@@ -58,13 +58,17 @@ private MavenArtifactIdentifier() { // static use only
   private static final ILog LOG = Platform.getLog(MavenArtifactIdentifier.class);
 
   public static Collection<ArtifactKey> identify(File classesLocation) {
+    return identify(classesLocation, true);
+  }
+
+  public static Collection<ArtifactKey> identify(File classesLocation, boolean searchCentral) {
```

Comment:
I don't thing such a parameter is useful here.
It should either be disabled for all cases or controlled by a preference, so callers don't have to individually decide what to pass here.

Reply:
But the callers have different use-cases. Some call it only on demand (i.e. the 'old' callers) and call it in advance to get the result for 'everything'. 
So I think it makes sense to distinguish here. If you think it is better the flag can also be renamed to 'fullSearch' or to an enum.

Reply:
The use case is always the same, and the caller should not need to know about the internals (e.g. if/that maven central might be queried), so I still think it would be better in general for a user to have the choice to query central at all, e.g. if I'm behind a company proxy or a like.

Reply:
Hmm, ok. Introduced a preference for that.